src/core/FilePath.cpp

```diff
@@ -211,6 +211,9 @@ FilePath::FilePath()
     else if (testSetDir(appDirPath + "/share")) {
     }
 #endif
+    // Last ditch test when running in the build directory (mainly for travis tests)
+    else if (testSetDir(QString(KEEPASSX_SOURCE_DIR) + "/share")) {
```

Comment:
Actually, Travis tests are meant to run on Debug build (before this commit https://github.com/keepassxreboot/keepassxc/commit/dee331d52d5df5b8b282eb8cfa7cbbc39ea77a62 they were) and the Release build should be used only by end users that want to install it or when building a release.
So the problem with icons and prefix shouldn't be there in the first place.

If you want to have multiple workspace for various dev/stable KeePassXC version you can do the following:
When compiling with **Debug** option: no problem, will load resource from SOURCE folder.
When compiling with **Release** option: install keepassxc to a destination folder with `make install DESTDIR=./dist` and execute that executable, not the one from `./build/src`

Those are my 2 cents, leaving this here won't hurt anyone but really it's useless.